waddrmgr/db.go

```diff
@@ -484,36 +790,86 @@ func putAccountInfo(tx walletdb.Tx, account uint32, encryptedPubKey,
 		acctType: actBIP0044,
 		rawData:  rawData,
 	}
-	return putAccountRow(tx, account, &acctRow)
+	if err := putAccountRow(tx, account, &acctRow); err != nil {
+		return err
+	}
+	// Update account id index
+	if err := putAccountIdIndex(tx, account, name); err != nil {
+		return err
+	}
+	// Update account name index
+	if err := putAccountNameIndex(tx, account, name); err != nil {
+		return err
+	}
+	return nil
 }
 
-// fetchNumAccounts loads the number of accounts that have been created from
-// the database.
-func fetchNumAccounts(tx walletdb.Tx) (uint32, error) {
-	bucket := tx.RootBucket().Bucket(acctBucketName)
+// putNumAllAddrs stores the provided metadata - number of new addrs - to the database.
+func putNumAllAddrs(tx walletdb.Tx, count uint32) error {
+	bucket := tx.RootBucket().Bucket(metaBucketName)
 
-	val := bucket.Get(acctNumAcctsName)
-	if val == nil {
-		str := "required num accounts not stored in database"
-		return 0, managerError(ErrDatabase, str, nil)
+	n, err := fetchNumAllAddresses(tx)
+	if err != nil {
+		return err
 	}
+	n += count
+	err = bucket.Put(numAllAddrsName, uint32ToBytes(n))
+	if err != nil {
+		str := fmt.Sprintf("failed to update metadata '%s'", numAllAddrsName)
+		return managerError(ErrDatabase, str, err)
+	}
+	return nil
+}
 
-	return binary.LittleEndian.Uint32(val), nil
+// putNumAccountAddrs stores the provided metadata - number of new account addrs - to the database.
+func putNumAccountAddrs(tx walletdb.Tx, account uint32, count uint32) error {
```

Comment:
The same is true for this function as `putNumAllAddrs`.  The function should take the value to write instead of a count to increment by.